index.js

```diff
@@ -34,34 +53,35 @@ module.exports = {
 
       var header = appConfig.contentSecurityPolicyHeader;
       var headerConfig = appConfig.contentSecurityPolicy;
-      var normalizedHost = options.host === '0.0.0.0' ? 'localhost' : options.host;
 
       if (options.liveReload) {
         ['localhost', '0.0.0.0'].forEach(function(host) {
-          headerConfig['connect-src'] = headerConfig['connect-src'] + ' ws://' + host + ':' + options.liveReloadPort;
-          headerConfig['script-src'] = headerConfig['script-src'] + ' ' + host + ':' + options.liveReloadPort;
+          var liveReloadHost = host + ':' + options.liveReloadPort;
+          headerConfig['connect-src'] += (' ' + 'ws://' + liveReloadHost);
+          headerConfig['script-src'] += (' ' + liveReloadHost);
         });
       }
 
-      if (header.indexOf('Report-Only')!==-1 && !('report-uri' in headerConfig)) {
-        headerConfig['connect-src'] = headerConfig['connect-src'] + ' http://' + normalizedHost + ':' + options.port + '/csp-report';
-        headerConfig['report-uri'] = 'http://' + normalizedHost + ':' + options.port + '/csp-report';
+      if (header.indexOf('Report-Only') !== -1 && !('report-uri' in headerConfig)) {
+        var normalizedHost = options.host === '0.0.0.0' ? 'localhost' : options.host;
+        var emberCliServer = 'http://' + normalizedHost + ':' + options.port + REPORT_PATH
+        headerConfig['connect-src'] += (' ' + emberCliServer);
```

Comment:
Also has to be `.push()`.

On a related note. Should the `REPORT_PATH` really be part of `connect-src`? While Chrome doesn't throw any errors at me, it's not listed as [allowed syntax](http://content-security-policy.com/#source_list).

Furthermore, at least in the new ember-cli beta, `options.host` is `undefined`, when no `--host` option is passed to `ember serve`. Ironically, [that was me](https://github.com/ember-cli/ember-cli/pull/4781).

All in all, this should be:

```js
      if (header.indexOf('Report-Only') !== -1 && !('report-uri' in headerConfig)) {
        var normalizedHost = options.host || 'localhost';
        var emberCliServer = 'http://' + normalizedHost + ':' + options.port;
        headerConfig['connect-src'].push(emberCliServer);
        headerConfig['report-uri'] = emberCliServer + REPORT_PATH;
      }
```